### DISCUSSION TITLE
Add meta data to seurat objects in a for loop

### DISCUSSION THREAD

**Discussion by dustin-mullaney** [satijalab/seurat#6415] [Q&A] [answered]
Hi,
Essentially I want to iterate over a list of seurat objects, and add meta data to each of them. However, I cannot get this to work. 
For instance, in the following example where objects is a list of seurat objects, the code runs but no meta data is added. However, if run the same assignment for each object manually, I can add the metadata without issue:

for (object in objects) { object[["new_column"]] <- "new_annotation" }

Am I missing something? I have tried several variations of this with no luck. 

Ultimately I would like to have a for loop with nested if statements so that I can loop over a list of objects, determine if each object meets a certain condition, and if it does add a column of metadata indicating so. Then when I integrate the objects in the list I can keep track of these conditions. I have successfully applied similar operations where I looped over a list of object names in string format, then used the get() function to call objects in the loop, that approach does not work in this situation though. Any suggestions/help is appreciated. 

Thanks

**Comment by AmelZulji** [comment, marked as answer]
Hi @dustin-mullaney,

The attempt is pretty close. You need to structure iteration slightly different. Here is the example of for loop which will work: 

```
for (object in names(objects)) {objects[[object]][["new_column"]] <- "new_annotation"}
```

**Reply by dustin-mullaney** [reply]
Thanks!
